lib/kmer_hash.hh

```diff
@@ -79,6 +79,135 @@ HashIntoType _hash_murmur(const std::string& kmer);
 HashIntoType _hash_murmur(const std::string& kmer,
                           HashIntoType& h, HashIntoType& r);
 HashIntoType _hash_murmur_forward(const std::string& kmer);
+
+class Kmer
+{
+
+public:
+
+    HashIntoType kmer_f, kmer_r, kmer_u;
+
+    Kmer(HashIntoType f, HashIntoType r, HashIntoType u)
+    {
+        kmer_f = f;
+        kmer_r = r;
+        kmer_u = u;
+    }
+	Kmer()
+	{
+		kmer_f = kmer_r = kmer_u = 0;
+	}
+
+    operator HashIntoType() const { return kmer_u; }
+	operator unsigned int() const { return kmer_u; }
+
+    bool operator< (const Kmer &other) const {
+        return kmer_u < other.kmer_u;
+    }
+
+    std::string get_string_rep(WordLength K) {
+        return _revhash(kmer_u, K);
+    }
+
+    const char * get_char_rep(WordLength K) {
+		std::string kmer_s = _revhash(kmer_u, K);
+        return kmer_s.c_str();
+    }
+};
+
+/*
+ * At first, I had many contructors defined on Kmer itself. However, it quickly
+ * became apparent that this would not work, as the need to pass in a uint
+ * for K made the calls ambiguous. Then I moved it to Traverser, but that
+ * really felt like I was shoehorning it in. So here we are, with a factory
+ * object. All apologies.
+ */
+class KmerFactory
+{
+protected:
+    WordLength _ksize;
+
+public:
+
+    KmerFactory(WordLength K): _ksize(K) {}
+
+    Kmer build_kmer(HashIntoType kmer_u)
```

Comment:
Please document that `kmer_u` mean the uniquified version; points for doxygen compatibility